python/rascaline/rascaline/utils/clebsch_gordan/_clebsch_gordan.py

```diff
@@ -34,79 +40,100 @@ def _standardize_keys(tensor: TensorMap) -> TensorMap:
-    selected_keys: Optional[Union[Labels, List[Labels]]] = None,
-    like=None,
+    selected_keys: Optional[Union[Labels, List[Union[Labels, None]]]] = None,
 ) -> List[Union[None, Labels]]:
     """
     Parses the `selected_keys` argument passed to public functions. Checks the
     values and returns a :py:class:`list` of :py:class:`Labels` objects, one for
-    each iteration of CG combination.
-
-    `like` is required if a new :py:class:`Labels` object is to be created by
-    :py:mod:`_dispatch`.
+    each iteration of CG combination.The `:param array_like:` determines the
+    array backend of the Labels created
     """
+    # Check the selected_keys
+    if (
+        (selected_keys is not None)
+        and (not isinstance(selected_keys, list))
+        and (not is_labels(selected_keys))
+    ):
+        raise TypeError(
+            "`selected_keys` must be `None`, `Labels` or List[Union[None, `Labels`]]"
+        )
+
+    if isinstance(selected_keys, list):
+        if not all(
+            [
+                is_labels(selected_keys[i]) or (selected_keys[i] is None)
+                for i in range(len(selected_keys))
+            ]
+        ):
+            raise TypeError(
+                "`selected_keys` must be a Labels or List[Union[None, Labels]]"
+            )
+
     # Check angular_cutoff arg
     if angular_cutoff is not None:
         if not isinstance(angular_cutoff, int):
             raise TypeError("`angular_cutoff` must be passed as an int")
         if angular_cutoff < 1:
             raise ValueError("`angular_cutoff` must be >= 1")
 
+    # we use a new variable for selected_keys so TorchScript can infer correct type
+    selected_keys_: List[Union[None, Labels]] = []
+
     if selected_keys is None:
         if angular_cutoff is None:  # no selections at all
-            selected_keys = [None] * n_iterations
+            selected_keys_ = [None] * n_iterations
         else:
             # Create a key selection with all angular channels <= the specified
             # angular cutoff
-            selected_keys = [
-                Labels(
-                    names=["spherical_harmonics_l"],
-                    values=_dispatch.int_range_like(
-                        0, angular_cutoff, like=like
-                    ).reshape(-1, 1),
-                )
-            ] * n_iterations
+            label = Labels(
+                names=["spherical_harmonics_l"],
+                values=_dispatch.int_array_like(
+                    list(range(0, angular_cutoff)), like=array_like
+                ).reshape(-1, 1),
+            )
+            selected_keys_ = [label] * n_iterations
 
-    if isinstance(selected_keys, Labels):
+    if is_labels(selected_keys):
         # Create a list, but only apply a key selection at the final iteration
-        selected_keys = [None] * (n_iterations - 1) + [selected_keys]
+        selected_keys_ = [None] * (n_iterations - 1)
+        selected_keys_.append(selected_keys)
```

Comment:
remain from making this TorchScript compatible
```suggestion
        selected_keys_ = [None] * (n_iterations - 1) + [selected_keys]
```